craft_application/services/testing.py

```diff
@@ -64,18 +64,20 @@ def process_spread_yaml(self, dest: pathlib.Path) -> None:
         emit.trace(f"Writing processed spread file to {dest}")
         spread_yaml.to_yaml_file(dest)
 
-    def run_spread(self, project_path: pathlib.Path) -> None:
+    def run_spread(self, spread_yaml: pathlib.Path) -> None:
```

Comment:
unrelated to this PR but it feels like a minor design issue how you need to call `process_spread_yaml(x)` and `run_spread(x)` with the same argument. Maybe the API should just be `run_spread(x)`?

Reply:
To run spread passing the directory seems correct since the implementation ignores the passed file path and only uses its parent directory.

Reply:
I like having a single method to run, but I would also like to have a way to run spread without pre-processing. Name suggestions welcome!